docs: introduce Mintlify-inspired Material theme (#2702)
codacy/docs: CONTRIBUTING.md

    mkdocs serve
    ```

### Working on the documentation theme

The documentation uses the installed `mkdocs-material` package with a small,
Codacy-owned theme layer in [`theme/`](theme/). This is

> the current theme source of truth. Do not copy, modify, or revive the legacy

Author: ```suggestion
the current theme source of truth. Don't copy, modify, or renew the legacy
```
